python/chronos/src/bigdl/chronos/data/tsdataset.py

```diff
@@ -812,3 +819,39 @@ def _check_basic_invariants(self, strict_check=False):
         # check dt sorted
         if strict_check:
             _check_dt_is_sorted(self.df, self.dt_col)
+
+    def get_cycle_length(self, aggregate='mode', top_k=3):
+        """
+        Calculate the cycle length of the time series in this TSDataset.
+
+        Args:
+            top_k (int): The freq with top top_k power after fft will be
+                used to check the autocorrelation.
+            aggregate (str): Select the mode of calculation time period,
+                We only support 'min', 'max', 'mode', 'median', 'mean'.
+
+        Returns:
+            A best cycle length.
+        """
+        assert isinstance(top_k, int),\
+            f"top_k type must be int, but found {type(top_k)}."
+        assert isinstance(aggregate, str),\
+            f"aggregate type must be str, but found {type(aggregate)}."
+        assert aggregate.lower().strip() in ['min', 'max', 'mode', 'median', 'mean'], \
+            f"We Only support 'min' 'max' 'mode' 'median' 'mean', but found {aggregate}."
+
+        res = self.df.groupby(self.id_col).apply(lambda x: [cycle_length_est(x[col].values, top_k)
+                                                 for col in self.target_col][0])
```

Comment:
I just reviewed this formula, and I found that this is kind of strange. you calculated each sub_df's cycle length and you take the first col's value (`[0]`) as this sub_df's cycle length?

I think the proper method is to calculate (id_num*target_num) cycle length and aggregate them.